llvm/lib/Transforms/Vectorize/VPlanConstruction.cpp

```diff
@@ -491,12 +480,34 @@ void VPlanTransforms::prepareForVectorization(VPlan &Plan, Type *InductionTy,
   VPBlockUtils::insertBlockAfter(VecPreheader, Plan.getEntry());
 
   VPBasicBlock *MiddleVPBB = Plan.createVPBasicBlock("middle.block");
-  VPBlockUtils::connectBlocks(LatchVPB, MiddleVPBB);
-  LatchVPB->swapSuccessors();
+  // The canonical LatchVPB has the header block as last successor. If it has
+  // another successor, this successor is an exit block - insert middle block on
+  // its edge. Otherwise, add middle block as another successor retaining header
+  // as last.
```

Comment:
Hmm, `prepareForVectorization()` is called after `buildPlainCFG()` (for both native and non-native paths), and with the above changes, `buildPlainCFG()` connects the latch to both header and exit blocks, i.e., should LatchVPB always have two successors now?